modules/urm/impl/src/main/java/com/innowise/educationalsystem/client/impl/EmailNotificationClient.java

```diff
@@ -0,0 +1,109 @@
+package com.innowise.educationalsystem.client.impl;
+
+import com.innowise.educationalsystem.client.NotificationClient;
+import com.innowise.educationalsystem.dto.mail.*;
+import com.innowise.educationalsystem.entity.Invite;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.support.Acknowledgment;
+import org.springframework.kafka.support.KafkaHeaders;
+import org.springframework.messaging.handler.annotation.Header;
+import org.springframework.messaging.handler.annotation.Payload;
+import org.springframework.stereotype.Component;
+import org.testcontainers.shaded.com.google.common.primitives.Longs;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class EmailNotificationClient implements NotificationClient {
+    private static final String SUBSCRIPTION_ID_HEADER = "subscription_id";
+    private static final String MAIL_TYPE = "INVITE";
+
+    @Value("${kafka.topic.mail-request}")
+    private String mailRequestTopic;
+
+    @Value("${kafka.topic.attempt-request}")
+    private String attemptRequestTopic;
+
+    private final KafkaTemplate<String, Object> kafkaTemplate;
+
+    private final Map<String, Invite> inviteMap = new HashMap<>();
+
+    @Override
+    public void sendInvite(Invite invite) {
+        inviteMap.put(invite.getId(), invite);
+        sendMailRequest(invite);
+    }
+
+    @KafkaListener(topics = "${kafka.topic.mail-response}",
+                   containerFactory = "kafkaListenerContainerFactory")
+    private void listenMailResponse(
+            @Payload MailResponse mailResponse,
+            @Header(KafkaHeaders.CORRELATION_ID) String correlationId,
+            Acknowledgment acknowledgment) {
+        Invite invite = inviteMap.get(correlationId);
+        if (invite == null) {
+            return;
+        }
+
+        if (mailResponse.getCreationStatus() == MailStatus.FAILED) {
+            sendMailRequest(invite);
```

Comment:
Will the service be trying reinviting forever? IF so, lets do a configurable amount of retries